From 4d8519c7acba6973c1e786d6899e71d8c828252e Mon Sep 17 00:00:00 2001
From: Jenny TC <jenny.tc@intel.com>
Date: Mon, 27 Oct 2014 09:17:33 +0530
Subject: [PATCH 1/1] WCOVE GPADC: wakeup GPADC sample wait queue

On receiving the GPADC interrupt, wakeup the wait queue, so
that the sampling routine can process the GPADC result register.

Change-Id: I3b8ae6a53a82dbad0db3b31f544a1234b73dc1a9
Signed-off-by: Jenny TC <jenny.tc@intel.com>
---
 drivers/iio/adc/intel_wcove_gpadc.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/iio/adc/intel_wcove_gpadc.c b/drivers/iio/adc/intel_wcove_gpadc.c
index 03650fac38d1..c7b933c869a1 100644
--- a/drivers/iio/adc/intel_wcove_gpadc.c
+++ b/drivers/iio/adc/intel_wcove_gpadc.c
@@ -127,8 +127,9 @@ static irqreturn_t gpadc_threaded_isr(int irq, void *data)
 {
 	struct gpadc_info *info = iio_priv(data);
 	struct gpadc_regs_t *regs = info->gpadc_regs;
-	/* Clear IRQLVL1MASK */
-	intel_soc_pmic_clearb(regs->mirqlvl1, regs->mirqlvl1_adc);
+
+	info->sample_done = 1;
+	wake_up(&info->wait);
 
 	return IRQ_HANDLED;
 }
-- 
1.7.9.5

